### ISSUE TITLE
Integration test timeouts waiting for dc

### ISSUE THREAD

**Issue by jeffbanks** [k8ssandra/k8ssandra#1020]
**Issue**
CassandraDatacenters never reach a valid state in two integration tests.

**Describe the bug**
During execution of integration tests for k8ssandra, two of the tests exhibit the following timeout behavior.

**integration-tests (3.11.10, TestRestoreAfterUpgrade)**
```
2021/08/09 18:28:35 Checking cassandradatacenter dc1 state in namespace k8ssandra2021080918085907...
    integration_steps.go:276: 
        Timed out after 900.000s.
        Expected
            <bool>: false
        to be true
2021/08/09 18:29:05 Installing and starting k8ssandra took 15m16.053640166s
2021/08/09 18:29:05 Not cleaning up cluster as requested
```

**integration-tests (3.11.10, TestUpgradeScenario)**
```
2021/08/09 18:35:46 Checking cassandradatacenter dc1 state in namespace k8ssandra2021080918175707...
    integration_steps.go:276: 
        Timed out after 900.000s.
        Expected
            <bool>: false
        to be true
2021/08/09 18:36:16 Installing and starting k8ssandra took 15m15.774429268s
2021/08/09 18:36:16 Not cleaning up cluster as requested
--- FAIL: TestUpgradeScenario (1686.72s)
    --- PASS: TestUpgradeScenario/Upgrade_from_v1.0.0 (508.25s)
    --- FAIL: TestUpgradeScenario/Upgrade_from_latest (1098.33s)
FAIL
FAIL	github.com/k8ssandra/k8ssandra/tests/integration	1687.209s
FAIL
Error: make: *** [Makefile:43: integ-test] Error 1
Error: Process completed with exit code 2.
```

**To Reproduce**
Push branch PR up to commit: dc193c0799f473f57b26f47745c739881c9d2c59

Example:
[Integration Test Run 1](https://github.com/k8ssandra/k8ssandra/runs/3283363266?check_suite_focus=true)
[Integration Test Run 2](https://github.com/k8ssandra/k8ssandra/pull/1013)

**Expected behavior**
Tests pass.

**Environment (please complete the following information):**
Local integration testing runs and GitHub environment.

**Helm charts version info**
```
k8ssandra       default         1               2021-08-09 14:45:33.6203166 -0500 CDT   deployed        k8ssandra-1.4.0-SNAPSHOT
minio           minio           1               2021-08-09 12:53:43.2373172 -0500 CDT   deployed        minio-8.0.10                    master
traefik         traefik         1               2021-08-09 12:53:42.2152218 -0500 CDT   deployed        traefik-9.19.2                  2.4.8
```

**Helm charts user-supplied values**
null

**Kubernetes version information:**

```
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.4", GitCommit:"e87da0bd6e03ec3fea7933c4b5263d151aafd07c", GitTreeState:"clean", BuildDate:"2021-02-18T16:12:00Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2", GitCommit:"faecb196815e248d3ecfb03c680a4507229c2a56", GitTreeState:"clean", BuildDate:"2021-01-21T01:11:42Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
```

**Kubernetes cluster kind:**
`kind`

**Comment by jsanda** [comment]
I just ran the `TestUpgradeScenario` scenario locally. For reference I ran it with this cmd line:

```
make integ-test TESTS=TestUpgradeScenario GO_FLAGS="-v" CLUSTER_CLEANUP="success"
```

I think it is a legit failure. Here is the exception reported in the Cassandra log:

```
ERROR [main] 2021-08-16 21:56:18,179 CassandraDaemon.java:909 - Fatal configuration error
org.apache.cassandra.exceptions.ConfigurationException: Cannot change the number of tokens from 256 to 16
	at org.apache.cassandra.service.StorageService.joinTokenRing(StorageService.java:1068)
	at org.apache.cassandra.service.StorageService.joinTokenRing(StorageService.java:1017)
	at org.apache.cassandra.service.StorageService.initServer(StorageService.java:799)
	at org.apache.cassandra.service.StorageService.initServer(StorageService.java:729)
	at org.apache.cassandra.service.CassandraDaemon.setup(CassandraDaemon.java:420)
	at org.apache.cassandra.service.CassandraDaemon.activate(CassandraDaemon.java:763)
	at org.apache.cassandra.service.CassandraDaemon.main(CassandraDaemon.java:887)
```
For Cassandra 3.11.x the default for `num_tokens` is 256. For 4.0 it is 16. The template that configures `num_tokens` needs to be updated to check if it is an upgrade vs an install. If it is an upgrade we shouldn't modify `num_tokens`. I will create a separate issue for this.

**Comment by jsanda** [comment]
Closed by #1031
